packages/expo-cli/src/commands/eject/Eject.js

```diff
@@ -101,241 +79,180 @@ export async function ejectAsync(projectRoot: string, options) {
+  /**
+   * Customize app.json
+   */
+  let { displayName, name } = await getAppNamesAsync(projectRoot);
+  appJson.displayName = displayName;
+  appJson.name = name;
 
-    // React Native 0.59 (shipped with sdk 33) adds @babel/runtime to dependencies
-    if (Versions.lteSdkVersion(exp, '32.0.0')) {
-      pkgJson.dependencies['@babel/runtime'] = '^7.0.0';
-    }
-    pkgJson.dependencies['react-native'] = reactNativeVersion;
-
-    if (pkgJson.jest !== undefined) {
-      newDevDependencies.push('jest');
-
-      if (pkgJson.jest.preset === 'jest-expo') {
-        pkgJson.jest.preset = 'react-native';
-      } else {
-        log(
-          `${chalk.bold(
-            'Warning'
-          )}: it looks like you've changed the Jest preset from jest-expo to ${
-            pkgJson.jest.preset
-          }. We recommend you make sure this Jest preset is compatible with ejected apps.`
-        );
-      }
-    }
+  if (appJson.expo.entryPoint && appJson.expo.entryPoint !== EXPO_APP_ENTRY) {
+    log(
+      chalk.yellow(`expo.entryPoint is already configured, we recommend using "${EXPO_APP_ENTRY}`)
+    );
+  } else {
+    appJson.expo.entryPoint = EXPO_APP_ENTRY;
+  }
 
-    // no longer relevant to an ejected project (maybe build is?)
-    delete pkgJson.scripts.eject;
+  log('Writing app.json...');
+  await fse.writeFile(path.resolve('app.json'), JSON.stringify(appJson, null, 2));
+  log(chalk.green('Wrote to app.json, please update it manually in the future.'));
 
-    log(`Updating your ${npmOrYarn} scripts in package.json...`);
+  /**
+   * Extract the template and copy it over
+   */
+  try {
+    let tempDir = tmp.dirSync();
+    await Exp.extractTemplateAppAsync(templateSpec, tempDir.name, appJson);
+    fse.copySync(path.join(tempDir.name, 'ios'), path.join(projectRoot, 'ios'));
+    fse.copySync(path.join(tempDir.name, 'android'), path.join(projectRoot, 'android'));
+    tempDir.removeCallback();
+    log('Successfully copied template native code.');
+  } catch (e) {
+    log(chalk.red(e.message));
+    log(chalk.red(`Eject failed, see above output for any issues.`));
+    log(chalk.yellow('You may want to delete the `ios` and/or `android` directories.'));
+    process.exit(1);
+  }
+
+  log(`Updating your ${npmOrYarn} scripts in package.json...`);
+  if (!pkgJson.scripts) {
+    pkgJson.scripts = {};
+  }
+  delete pkgJson.scripts.eject;
+  pkgJson.scripts.start = 'react-native start';
+  pkgJson.scripts.ios = 'react-native run-ios';
+  pkgJson.scripts.android = 'react-native run-android';
+
+  const { sdkVersion } = exp;
+  const versions = await Versions.versionsAsync();
+  const reactNativeVersion = versions.sdkVersions[sdkVersion].facebookReactNativeVersion;
+  pkgJson.dependencies['react-native'] = reactNativeVersion;
 
-    await fse.writeFile(path.resolve('package.json'), JSON.stringify(pkgJson, null, 2));
+  // TODO: how should we version react-native-unimodules to match up with react-native version?
```

Comment:
actually this should be easy to find by looking at the version from the package.json in the bare project